### ISSUE TITLE
Docker error with version 0.6.0+

### ISSUE THREAD

**Issue by Arkessler** [fern-api/fern-python#407]
Hello! I'm currently hitting a Docker issue when attempting to generate Python SDKs with any version of this repo past 0.5.0-rc2. 
I've tried with `0.6.0` - `0.6.2` and installed `poetry` to see if it would help. `poetry` didn't seem to make a difference. 

As an interesting oddity, my fern version is `0.14.3`
```
fern --version
0.14.3
```
But all of the generation seems to use fern `0.14.1`, while printing that "an upgrade is available". Not sure if there's some kind of fern version override built in to the different generator versions.

Running with 
```
just-python:
    generators:
      - name: fernapi/fern-python-sdk
        version: 0.5.0-rc2
        output:
          location: local-file-system
```
Gives:
```
DEBUG 2023-11-06T20:42:10.270Z Running fern (fern-api@0.14.1)
DEBUG 2023-11-06T20:42:10.995Z [management]: fernapi/fern-python-sdk Migrating IR from v22 to v21
DEBUG 2023-11-06T20:42:10.995Z [management]: fernapi/fern-python-sdk Migrating IR from v21 to v20
DEBUG 2023-11-06T20:42:12.579Z [management]: fernapi/fern-python-sdk Wrote IR to: /private/var/folders/hj/10ls64s54t70qw25nxhnc62m0000gn/T/fern-20176-3wHTIjfoiqZF/tmp-20176-ObBxKtoV8kQL
DEBUG 2023-11-06T20:42:12.579Z [management]: fernapi/fern-python-sdk Will write config.json to: /private/var/folders/hj/10ls64s54t70qw25nxhnc62m0000gn/T/fern-20176-3wHTIjfoiqZF/tmp-20176-nii4ruMujmJg
DEBUG 2023-11-06T20:42:12.579Z [management]: fernapi/fern-python-sdk Will write output to: /private/var/folders/hj/10ls64s54t70qw25nxhnc62m0000gn/T/fern-20176-3wHTIjfoiqZF/tmp-20176-tryfynK5dfme
DEBUG 2023-11-06T20:42:16.892Z [management]: fernapi/fern-python-sdk rm -rf /Users/alexi/python-sdk/client/management
DEBUG 2023-11-06T20:42:16.894Z [management]: fernapi/fern-python-sdk Copying generated files to /Users/alexi/python-sdk/client/management
INFO  2023-11-06T20:42:16.964Z [management]: fernapi/fern-python-sdk Wrote files to /Users/alexi/python-sdk/client/management
DEBUG 2023-11-06T20:42:16.965Z Checking if fern-api upgrade is available...
DEBUG 2023-11-06T20:42:17.165Z Latest version: 0.14.3. Upgrade available.
INFO  2023-11-06T20:42:17.170Z                                                                                                                    ╭───────────────────────────────────────╮
                                                                                                                                                  │                                       │
                                                                                                                                                  │   Update available 0.14.1 → 0.14.3    │
                                                                                                                                                  │      Run fern upgrade to update       │
                                                                                                                                                  │                                       │
                                                                                                                                                  ╰───────────────────────────────────────╯

┌─
│ ✓  fernapi/fern-python-sdk
└─
```

Running with 
```
just-python:
    generators:
      - name: fernapi/fern-python-sdk
        version: 0.6.0
        output:
          location: local-file-system
```
Gives:
```
DEBUG 2023-11-06T20:36:16.715Z Running fern (fern-api@0.14.1)
DEBUG 2023-11-06T20:36:17.409Z [management]: fernapi/fern-python-sdk Migrating IR from v22 to v21
DEBUG 2023-11-06T20:36:17.409Z [management]: fernapi/fern-python-sdk Migrating IR from v21 to v20
DEBUG 2023-11-06T20:36:18.972Z [management]: fernapi/fern-python-sdk Wrote IR to: /private/var/folders/hj/10ls64s54t70qw25nxhnc62m0000gn/T/fern-19745-3R7Fu2m39zQb/tmp-19745-aQhImV2LEdt5
DEBUG 2023-11-06T20:36:18.973Z [management]: fernapi/fern-python-sdk Will write config.json to: /private/var/folders/hj/10ls64s54t70qw25nxhnc62m0000gn/T/fern-19745-3R7Fu2m39zQb/tmp-19745-q6wdAsusJyNq
DEBUG 2023-11-06T20:36:18.973Z [management]: fernapi/fern-python-sdk Will write output to: /private/var/folders/hj/10ls64s54t70qw25nxhnc62m0000gn/T/fern-19745-3R7Fu2m39zQb/tmp-19745-VjD7Q5KOdrjt
ERROR 2023-11-06T20:36:20.103Z [management]: fernapi/fern-python-sdk Docker exited with a non-zero exit code.
DEBUG 2023-11-06T20:36:20.153Z [management]: fernapi/fern-python-sdk Error: Docker exited with a non-zero exit code.
                                                                         at tryRunDocker (/Users/alexi/.npm/_npx/db1ce26fd8597fe2/node_modules/fern-api/bundle.cjs:544827:11)
                                                                         at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
                                                                         at async runDocker (/Users/alexi/.npm/_npx/db1ce26fd8597fe2/node_modules/fern-api/bundle.cjs:544801:5)
                                                                         at async runGenerator (/Users/alexi/.npm/_npx/db1ce26fd8597fe2/node_modules/fern-api/bundle.cjs:544895:3)
                                                                         at async writeFilesToDiskAndRunGenerator (/Users/alexi/.npm/_npx/db1ce26fd8597fe2/node_modules/fern-api/bundle.cjs:544951:3)
                                                                         at async /Users/alexi/.npm/_npx/db1ce26fd8597fe2/node_modules/fern-api/bundle.cjs:544914:9
                                                                         at async TaskContextImpl.runInteractiveTask (/Users/alexi/.npm/_npx/db1ce26fd8597fe2/node_modules/fern-api/bundle.cjs:530059:7)
                                                                         at async Promise.all (index 0)
                                                                         at async runLocalGenerationForWorkspace (/Users/alexi/.npm/_npx/db1ce26fd8597fe2/node_modules/fern-api/bundle.cjs:544909:19)
                                                                         at async generateWorkspace (/Users/alexi/.npm/_npx/db1ce26fd8597fe2/node_modules/fern-api/bundle.cjs:545702:5)
```

Would appreciate any tips on how to further debug, I'm not sure how to get a closer look at what's going wrong in Docker other than the `--log-level debug` I'm currently running with.

**Comment by dsinghvi** [comment]
@Arkessler if you run `fern upgrade` and use the latest generator things should work -- we've released the latest version of fern with fixes!

**Comment by Arkessler** [comment]
Thank you for the fixes and advice @dsinghvi ! Apologies it took so long to figure out on my side, had some other work items take priority.

For anyone that comes across this in the future, the underlying issue was actually due to 
> As an interesting oddity, my fern version is 0.14.3

> fern --version
> 0.14.3
> But all of the generation seems to use fern 0.14.1, while printing that "an upgrade is available". Not sure if there's some kind of fern version override built in to the different generator versions.

Because of how our Makefile targets chain together, `fern generate` gets run a couple directories over from where the actual api files are. I saw the "upgrade your fern version" message and ran `fern upgrade`, but didn't realize that command was directory specific. 

When I ran `fern upgrade`, all it did was upgrade my local binary, thus:
> fern --version
> 0.14.3

But the generator was still calling the older version of fern associated with the un-upgraded fern spec files, leading to incompatibilities between the fern version and the generator version.

After running `fern upgrade` in the directory where we keep our fern spec, everything worked as intended.

(That said, --keepDocker did help and was a valuable debugging tool, thanks!)
